src/library/scala/collection/SeqView.scala

```diff
@@ -30,16 +30,42 @@ trait SeqView[+A] extends SeqOps[A, View, View[A]] with View[A] {
   def concat[B >: A](suffix: SeqView.SomeSeqOps[B]): SeqView[B] = new SeqView.Concat(this, suffix)
   def appendedAll[B >: A](suffix: SeqView.SomeSeqOps[B]): SeqView[B] = new SeqView.Concat(this, suffix)
   def prependedAll[B >: A](prefix: SeqView.SomeSeqOps[B]): SeqView[B] = new SeqView.Concat(prefix, this)
+
+  override def sorted[B >: A](implicit ord: Ordering[B]): SeqView[A] = {
+    val len = this.length // does not evaluate any view operations; checks `underlying.length`
```

Comment:
`underlying.length` does evaluate the view contents, in case the underlying collection is a `List`, for instance. You should probably use `knownSize`.

Reply:
we need the exact size to create the array. the other option is to use `knownSize` and fall back to an `ArrayBuffer`, but it's not clear to me that that's more efficient.